moped-etl/arcgis/components_to_agol.py

```diff
@@ -54,74 +63,192 @@ def make_esri_feature(*, esri_geometry_key, geometry, attributes):
     return feature
 
 
-def main():
+def main(args):
     logger.info("Getting token...")
     get_token()
 
-    logger.info("Downloading component data...")
-    data = make_hasura_request(query=COMPONENTS_QUERY)["component_arcgis_online_view"]
-
-    logger.info(f"{len(data)} component records to process")
-
-    # lines and points must be stored in different layers in AGOL
-    all_features = {"lines": [], "points": [], "combined": []}
-
-    logger.info("Building Esri feature objects...")
-    for component in data:
-        # extract geometry and line geometry from component data
-        # for line features, the line geometry is redundant/identical to geometry
-        # for point features, it is the buffered ring around the point as defined
-        # in the Moped component view
-        geometry = component.pop("geometry")
-        line_geometry = component.pop("line_geometry")
-
-        if not geometry:
-            continue
+    if args.full:
```

Comment:
this first block is the main content of the previous version of the script. It deletes and replaces all component features. I think it would be handy to have around as an option.